### ISSUE TITLE
Profiling build of rules_haskell no longer works

### ISSUE THREAD

**Issue by mboes** [tweag/rules_haskell#324]
The build using the `dbg` configuration fails currently:

```
$ cd rules_haskell
$ nix-shell --pure
$ bazel build -c dbg //...
INFO: Build options have changed, discarding analysis cache.
DEBUG: /home/mboes/.cache/bazel/_bazel_mboes/822aa44857e17f47d08bb198503f3032/external/io_bazel_rules_go/go/private/rules/binary.bzl:54:5: DEPRECATED: linkstamp, please use x_def for all stamping now main
INFO: Analysed 144 targets (0 packages loaded).
INFO: Found 144 targets...
INFO: From Executing genrule @com_github_bazelbuild_buildtools//build:parse.y.go_yacc:

conflicts: 58 shift/reduce
ERROR: /home/mboes/src/tweag/rules_haskell/tools/runfiles/BUILD:8:1: HaskellBuildLibrary //tools/runfiles:runfiles failed (Exit 1)

<interactive>:1:1: error:
    Failed to load interface for `GHC.Integer.Type'
    Perhaps you haven't installed the "p_dyn" libraries for package `integer-gmp-1.0.1.0'?
    Use -v to see a list of the files searched for.
INFO: Elapsed time: 3.295s, Critical Path: 0.76s
INFO: 3 processes: 1 local, 2 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

**Comment by mrkkrp** [comment]
To summarize:

* GHC bindists typically do not contain libraries compiled in `p_dyn` way.
* Thus `-prof` with `-dynamic` or `-dynamic-too` is not going to work.
* Cabal just doesn't compile dynamic versions of things when `-prof` is enabled.

So the solution seems to be: do the same thing as Cabal does and skip creation of dynamic libraries when profiling is enabled.

BTW, @Fuuzetsu do you have any thoughts?

**Comment by Fuuzetsu** [comment]
I'd copy cabal and enquire into why GHC bindist doesn't ship with p_dyn.

**Comment by mrkkrp** [comment]
Resolved by #332.
